json_tests/tests/test_json.rs

```diff
@@ -675,26 +679,37 @@ fn test_parse_unusual_ok<T>(errors: Vec<(&str, T)>)
     for (s, value) in errors {
         let v: T = from_str(s).unwrap();
         assert_eq!(v, value.clone());
+
+        let v: T = from_iter(s.bytes().map(Ok)).unwrap();
+        assert_eq!(v, value.clone());
     }
 }
 
 // FIXME (#5527): these could be merged once UFCS is finished.
 fn test_parse_err<T>(errors: Vec<(&'static str, Error)>)
     where T: Debug + PartialEq + de::Deserialize,
 {
-    for (s, err) in errors {
-        match (err, from_str::<T>(s).unwrap_err()) {
+    for &(s, ref err) in &errors {
```

Comment:
This is because `Error` is not `Copy` or `Clone` so I need a reference in order to test both the error from `from_str` and the one from `from_iter`.